src/Text/Regex/Base/RegexLike.hs

```diff
@@ -115,54 +136,62 @@ class (RegexOptions regex compOpt execOpt) => RegexMaker regex compOpt execOpt s
   makeRegexOpts c e s = maybe (error "makeRegexOpts failed") id (makeRegexOptsM c e s)
   makeRegexOptsM c e s = return (makeRegexOpts c e s)
 
-----------------
+
 -- | RegexLike is parametrized on a regular expression type and a
 -- source type to run the matching on.
 --
--- There are default implementations: matchTest and matchOnceText use
--- matchOnce; matchCount and matchAllText use matchAll. matchOnce uses
--- matchOnceText and matchAll uses matchAllText. So a minimal complete
--- instance need to provide at least (matchOnce or matchOnceText) and
--- (matchAll or matchAllText).  Additional definitions are often
+-- There are default implementations: 'matchTest' and 'matchOnceText' use
+-- 'matchOnce'; 'matchCount' and 'matchAllText' use 'matchAll'.
+-- Conversely, 'matchOnce' uses
+-- 'matchOnceText' and 'matchAll' uses 'matchAllText'. So a minimal complete
+-- instance need to provide at least ('matchOnce' or 'matchOnceText') and
+-- ('matchAll' or 'matchAllText').  Additional definitions are often
 -- provided where they will increase efficiency.
 --
 -- > [ c | let notVowel = makeRegex "[^aeiou]" :: Regex, c <- ['a'..'z'], matchTest notVowel [c]  ]
 -- >
 -- > "bcdfghjklmnpqrstvwxyz"
 --
 -- The strictness of these functions is instance dependent.
-class (Extract source)=> RegexLike regex source where
+--
+class (Extract source) => RegexLike regex source where
+
   -- | This returns the first match in the source (it checks the whole
   -- source, not just at the start). This returns an array of
   -- (offset,length) index pairs for the match and captured
   -- substrings.  The offset is 0-based.  A (-1) for an offset means a
   -- failure to match.  The lower bound of the array is 0, and the 0th
   -- element is the (offset,length) for the whole match.
-  matchOnce  :: regex -> source-> Maybe MatchArray
-  -- | matchAll returns a list of matches.  The matches are in order
+  matchOnce  :: regex -> source -> Maybe MatchArray
+
+  -- | @matchAll@ returns a list of matches.  The matches are in order
   -- and do not overlap. If any match succeeds but has 0 length then
   -- this will be the last match in the list.
-  matchAll   :: regex -> source-> [MatchArray]
-  -- | matchCount returns the number of non-overlapping matches
-  -- returned by matchAll.
-  matchCount :: regex -> source-> Int
-  -- | matchTest return True if there is a match somewhere in the
+  matchAll   :: regex -> source -> [MatchArray]
+
+  -- | @matchCount@ returns the number of non-overlapping matches
+  -- returned by @matchAll@.
+  matchCount :: regex -> source -> Int
+
+  -- | @matchTest@ returns @True@ if there is a match somewhere in the
   -- source (it checks the whole source not just at the start).
-  matchTest  :: regex -> source-> Bool
-  -- | This is matchAll with the actual subsections of the source
+  matchTest  :: regex -> source -> Bool
+
+  -- | This is @matchAll@ with the actual subsections of the source
```

Comment:
```suggestion
  -- | This is 'matchAll' with the actual subsections of the source
```

Reply:
Uumh, this is the description text for `matchAll`, why would you want to insert a link here (to itself?).

Reply:
Ah, i tend to link these. Stylistic w/e.